<a href="https://www.kaggle.com/code/ayushs9020/understanding-the-competition-s3e19?scriptVersionId=136346937" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

# Forecasting Mini-Course Sales 💰

<IMG SRC = "https://i0.wp.com/laughbreak.com/wp-content/uploads/2019/09/Banana-Ducks.jpg?fit=855%2C1024&ssl=1" WIDTH = 400>
    
**I could not downalod the actual image from playground series, so I am using the banana Duck, hope you will not mind**
    
$Kaggle$ $Mini-Courses$ are `free`/`self-paced` courses that `teach` you the `essential skills` for `Data Science`/`Machine Learning`. They are designed to be `accessible to beginners`, but they also offer `challenges`/`exercises` that will help you grow your skills. Each mini-course is divided into `several modules`, each of which covers a `specific topic`. The modules are typically `short`/`easy` to follow, and they include `interactive exercises` that allow you to practice what you've learned. At the `end` of each mini-course, you will be able to complete a `project` that demonstrates your mastery of the material. These projects are a great way to `showcase the skills` to potential employers or clients.

It becomes very `important`/`profitable` both with `time`/`resources` for the `Kaggle Team` to `predict the Sales` of a `particular course` `before` its `launch` and `major development` which helps them in various after tasks such as 

* **Optimize Marketing Campaigns** - By knowing `how many people` are `likely` to `purchase a mini-course`. Kaggle can allocate its `marketing budget` more `effectively`. For example, if a `mini-course` is `expected to be very popular`, Kaggle can `invest more in advertising` it. Conversely, if a `mini-course` is `not expected to be very popular`, Kaggle can `save money by investing less` in advertising it.
* **Pricing** - Knowing `how many people` are `likely to purchase` a mini-course can help Kaggle `set a price` that is `both profitable and attractive` to buyers. For example, if a `mini-course is expected to be very popular`, Kaggle can `charge a higher price for it`. Conversely, if a `mini-course is not expected to be very popular`, Kaggle can `charge a lower price for it`.
* **Improve Course Experience**. By knowing `how many people` are `likely to purchase a mini-course`, Kaggle can `make sure` that the `mini-course` is `meeting the needs` of its users. For example, if a `mini-course is not attracting` as many buyers as expected, Kaggle can `review the content of the mini-course` to see if it can be improved.
* **New Opportunities** - By predicting Kaggle Mini-Course sales, Kaggle can `identify new opportunities` for growth. For example, if a `mini-course` is `unexpectedly popular`, Kaggle can consider `creating more` mini-courses on `similar topics`.

# 1 | Advisory 💡

# 2 | Data 📊

In [1]:
import pandas as pd 

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/playground-series-s3e19/sample_submission.csv
/kaggle/input/playground-series-s3e19/train.csv
/kaggle/input/playground-series-s3e19/test.csv


Now lets dive into the data

In [2]:
train = pd.read_csv("/kaggle/input/playground-series-s3e19/train.csv")
train

id        date    country         store  \
0            0  2017-01-01  Argentina  Kaggle Learn   
1            1  2017-01-01  Argentina  Kaggle Learn   
2            2  2017-01-01  Argentina  Kaggle Learn   
3            3  2017-01-01  Argentina  Kaggle Learn   
4            4  2017-01-01  Argentina  Kaggle Learn   
...        ...         ...        ...           ...   
136945  136945  2021-12-31      Spain     Kagglazon   
136946  136946  2021-12-31      Spain     Kagglazon   
136947  136947  2021-12-31      Spain     Kagglazon   
136948  136948  2021-12-31      Spain     Kagglazon   
136949  136949  2021-12-31      Spain     Kagglazon   

                                               product  num_sold  
0                    Using LLMs to Improve Your Coding        63  
1                        Using LLMs to Train More LLMs        66  
2       Using LLMs to Win Friends and Influence People         9  
3           Using LLMs to Win More Kaggle Competitions        59  
4                           Using LLMs to Write Better        49  
...                                                ...       ...  
136945               Using LLMs to Improve Your Coding       700  
136946                   Using LLMs to Train More LLMs       752  
136947  Using LLMs to Win Friends and Influence People       111  
136948      Using LLMs to Win More Kaggle Competitions       641  
136949                      Using LLMs to Write Better       539  

[136950 rows x 6 columns]

This is our `Training Data` containing $1,36,950$ `Rows` and $6$ columns, accounting to a total of $8,21,700$ values. This huge amount of data will help in the dense modelinng 

In [3]:
test = pd.read_csv("/kaggle/input/playground-series-s3e19/test.csv")
test

id        date    country         store  \
0      136950  2022-01-01  Argentina  Kaggle Learn   
1      136951  2022-01-01  Argentina  Kaggle Learn   
2      136952  2022-01-01  Argentina  Kaggle Learn   
3      136953  2022-01-01  Argentina  Kaggle Learn   
4      136954  2022-01-01  Argentina  Kaggle Learn   
...       ...         ...        ...           ...   
27370  164320  2022-12-31      Spain     Kagglazon   
27371  164321  2022-12-31      Spain     Kagglazon   
27372  164322  2022-12-31      Spain     Kagglazon   
27373  164323  2022-12-31      Spain     Kagglazon   
27374  164324  2022-12-31      Spain     Kagglazon   

                                              product  
0                   Using LLMs to Improve Your Coding  
1                       Using LLMs to Train More LLMs  
2      Using LLMs to Win Friends and Influence People  
3          Using LLMs to Win More Kaggle Competitions  
4                          Using LLMs to Write Better  
...                                               ...  
27370               Using LLMs to Improve Your Coding  
27371                   Using LLMs to Train More LLMs  
27372  Using LLMs to Win Friends and Influence People  
27373      Using LLMs to Win More Kaggle Competitions  
27374                      Using LLMs to Write Better  

[27375 rows x 5 columns]

This is our `Training Data` containing $27,375$ `Rows` and $5$ columns, accounting to a total of $1,36,875$ values

# 3 | TO DO LIST 📃️ 

```
TO DO 1 : EXPLORE THE DATA

TO DO 2 : VISUALIZE THE DATA

TO DO 3 : TRAIN SOME MODELS

TO DO 4 : PREPREOCESS THE DATA

TO DO 5 : TRAIN SOME NNs

TO DO 6 : ADD WANDB SUPPORT

TO DO 7 : DANCE
```

# 4 | Ending 💹

**THAT IT FOR TODAY GUYS**

**WE WILL GO DEEPER INTO THE DATA IN THE UPCOMING VERSIONS**

**PLEASE COMMENT YOUR THOUGHTS, HIHGLY APPRICIATED**

**DONT FORGET TO MAKE AN UPVOTE, IF YOU LIKED MY WORK  $:)$**

<img src = "https://i.imgflip.com/19aadg.jpg">

**PEACE OUT $!!!$**